botcore/perms.py

```diff
@@ -169,100 +137,75 @@ def is_admin_user(error=False):
     Cog must have bot as an instance variable.
 
     Args:
-        error: Whether to raise error if check fails.
-
-    Raises:
-        NotAdminUser: Check failed and error == True.
+        error: Whether to send error message if check fails.
     """
 
     def predicate(cog, ctx):
-        member = cog.bot.get_guild(config["server-id"]).get_member(ctx.author.id)
-        if set(config["admin-roles"]).isdisjoint(map(lambda r: r.id, member.roles)):
-            if error:
-                raise NotAdminUser("You are not authorised to do that.")
-            return False
-        return True
-    return check(predicate)
+        member = _get_member(cog, ctx.author.id)
+        if set(config["admin-roles"]).isdisjoint(_get_role_ids(member)):
```

Comment:
What is happening here? This is something that looks like could be extracted into a helper function that has a much clearer function name "is_user_admin(member)" or something like that